src/nvim/window.c

```diff
@@ -2135,6 +2149,17 @@ void win_close_othertab(win_T *win, int free_buf, tabpage_T *tp)
     }
   }
 
+  // fire WinClosed event just before freeing memory associated with window
+  if (has_event(EVENT_WINCLOSED)) {
+    apply_autocmds(EVENT_WINCLOSED, win->w_buffer->b_fname,
+                   win->w_buffer->b_fname, false, win->w_buffer);
+
+    // If autocommands closed the window, there's nothing else to do
+    if (!win_valid_any_tab(win)) {
```

Comment:
same comments as for the `win_close` case.

Reply:
The same comment as before